programs/drift/src/math/margin.rs

```diff
@@ -342,14 +344,17 @@ pub fn calculate_margin_requirement_and_total_collateral_and_liability_info(
                     &spot_market,
                     &strict_oracle_price,
                     Some(signed_token_amount),
-                    context.margin_type,
+                    if !margin_calc_valid && context.margin_type == MarginRequirementType::Initial {
```

Comment:
need to check if its a liquidation too?

Reply:
do you mean check if its a liquidator account trying to take over a position and let it skip this check?